File: dqops/src/main/java/com/dqops/execution/errorsampling/jobs/ErrorSamplerResult.java
/*
 * Copyright © 2021-Present DQOps, Documati sp. z o.o. (support@dqops.com)
 *
 * This file is licensed under the Business Source License 1.1,
 * which can be found in the root directory of this repository.
 *
 * Change Date: This file will be licensed under the Apache License, Version 2.0,
 * four (4) years from its last modification date.
 */
package com.dqops.execution.errorsampling.jobs;

import com.dqops.execution.errorsampling.ErrorSamplerExecutionSummary;
import com.dqops.utils.docs.generators.SampleValueFactory;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import io.swagger.annotations.ApiModel;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * Returns the result with the summary of the error samples collected.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@ApiModel(value = "ErrorSamplerResult", description = "Returns the result with the summary of the error samples collected.")
@EqualsAndHashCode(callSuper = false)
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@Data
public class ErrorSamplerResult {
    /**
     * The total count of all executed error samplers. This count only includes data quality checks that have an error sampling template defined.
     */
    @JsonPropertyDescription("The total count of all executed error samplers. This count only includes data quality checks that have an error sampling template defined. ")
    private int executedErrorSamplers;

    /**
     * The count of columns for which DQOps executed an error sampler and tried to collect error samples.
     */
    @JsonPropertyDescription("The count of columns for which DQOps executed an error sampler and tried to collect error samples.")
    private int columnsAnalyzed;

    /**
     * The count of columns for which DQOps managed to obtain error samples.
     */
    @JsonPropertyDescription("The count of columns for which DQOps managed to obtain error samples.")
    private int columnsSuccessfullyAnalyzed;

    /**
     * The count of error samplers that failed to execute.
     */
    @JsonPropertyDescription("The count of error samplers that failed to execute.")
    private int totalErrorSamplersFailed;

    /**
     * The total number of error samples (values) that were collected.
     */
    @JsonPropertyDescription("The total number of error samples (values) that were collected.")
    private int totalErrorSamplesCollected;

    /**
     * The default parameterless constructor.
     */
    public ErrorSamplerResult() {
    }

    /**
     * Creates a collect error samples job result from the execution summary.
     * @param errorSamplerExecutionSummary Error sampler execution summary.
     * @return The job result object.
     */
    public static ErrorSamplerResult fromErrorSamplerExecutionSummary(ErrorSamplerExecutionSummary errorSamplerExecutionSummary) {
        ErrorSamplerResult runChecksQueueJobResult = new ErrorSamplerResult() {{
            setExecutedErrorSamplers(errorSamplerExecutionSummary.getTotalCollectorsExecuted());
            setColumnsAnalyzed(errorSamplerExecutionSummary.getColumnsAnalyzedCount());
            setColumnsSuccessfullyAnalyzed(errorSamplerExecutionSummary.getColumnsSuccessfullyAnalyzed());
            setTotalErrorSamplersFailed(errorSamplerExecutionSummary.getTotalCollectorsFailed());
            setTotalErrorSamplesCollected(errorSamplerExecutionSummary.getTotalCollectedResults());
        }};

        return runChecksQueueJobResult;
    }

    public static class ErrorSamplerResultResultSampleFactory implements SampleValueFactory<ErrorSamplerResult> {
        @Override
        public ErrorSamplerResult createSample() {
            // Columns analyzed, 1*3 + 2*2 + 4*1
            // One of the sensors failed for a 3 statistics column.
            // One for one of the two 2-statistics columns.
            // One for one of the three 1-statistic columns.
            return new ErrorSamplerResult() {{
                setExecutedErrorSamplers(11);
                setColumnsAnalyzed(7);
                setColumnsSuccessfullyAnalyzed(4);
                setTotalErrorSamplersFailed(3);
                setTotalErrorSamplesCollected(8);
            }};
        }
    }
}

File: distribution/python/dqops/client/models/error_sampler_result.py
from typing import Any, Dict, List, Type, TypeVar, Union

from attrs import define as _attrs_define
from attrs import field as _attrs_field

from ..types import UNSET, Unset

T = TypeVar("T", bound="ErrorSamplerResult")


@_attrs_define
class ErrorSamplerResult:
    """Returns the result with the summary of the error samples collected.

    Attributes:
        executed_error_samplers (Union[Unset, int]): The total count of all executed error samplers. This count only
            includes data quality checks that have an error sampling template defined.
        columns_analyzed (Union[Unset, int]): The count of columns for which DQOps executed an error sampler and tried
            to collect error samples.
        columns_successfully_analyzed (Union[Unset, int]): The count of columns for which DQOps managed to obtain error
            samples.
        total_error_samplers_failed (Union[Unset, int]): The count of error samplers that failed to execute.
        total_error_samples_collected (Union[Unset, int]): The total number of error samples (values) that were
            collected.
    """

    executed_error_samplers: Union[Unset, int] = UNSET
    columns_analyzed: Union[Unset, int] = UNSET
    columns_successfully_analyzed: Union[Unset, int] = UNSET
    total_error_samplers_failed: Union[Unset, int] = UNSET
    total_error_samples_collected: Union[Unset, int] = UNSET
    additional_properties: Dict[str, Any] = _attrs_field(init=False, factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        executed_error_samplers = self.executed_error_samplers
        columns_analyzed = self.columns_analyzed
        columns_successfully_analyzed = self.columns_successfully_analyzed
        total_error_samplers_failed = self.total_error_samplers_failed
        total_error_samples_collected = self.total_error_samples_collected

        field_dict: Dict[str, Any] = {}
        field_dict.update(self.additional_properties)
        field_dict.update({})
        if executed_error_samplers is not UNSET:
            field_dict["executed_error_samplers"] = executed_error_samplers
        if columns_analyzed is not UNSET:
            field_dict["columns_analyzed"] = columns_analyzed
        if columns_successfully_analyzed is not UNSET:
            field_dict["columns_successfully_analyzed"] = columns_successfully_analyzed
        if total_error_samplers_failed is not UNSET:
            field_dict["total_error_samplers_failed"] = total_error_samplers_failed
        if total_error_samples_collected is not UNSET:
            field_dict["total_error_samples_collected"] = total_error_samples_collected

        return field_dict

    @classmethod
    def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
        d = src_dict.copy()
        executed_error_samplers = d.pop("executed_error_samplers", UNSET)

        columns_analyzed = d.pop("columns_analyzed", UNSET)

        columns_successfully_analyzed = d.pop("columns_successfully_analyzed", UNSET)

        total_error_samplers_failed = d.pop("total_error_samplers_failed", UNSET)

        total_error_samples_collected = d.pop("total_error_samples_collected", UNSET)

        error_sampler_result = cls(
            executed_error_samplers=executed_error_samplers,
            columns_analyzed=columns_analyzed,
            columns_successfully_analyzed=columns_successfully_analyzed,
            total_error_samplers_failed=total_error_samplers_failed,
            total_error_samples_collected=total_error_samples_collected,
        )

        error_sampler_result.additional_properties = d
        return error_sampler_result

    @property
    def additional_keys(self) -> List[str]:
        return list(self.additional_properties.keys())

    def __getitem__(self, key: str) -> Any:
        return self.additional_properties[key]

    def __setitem__(self, key: str, value: Any) -> None:
        self.additional_properties[key] = value

    def __delitem__(self, key: str) -> None:
        del self.additional_properties[key]

    def __contains__(self, key: str) -> bool:
        return key in self.additional_properties
